### DISCUSSION TITLE
Dynamic inline JS scripts not working anymore

### DISCUSSION THREAD

**Discussion by morceaudebois** [livewire/livewire#8613] [1. Help] [answered]
Hi! I recently updated my project from Livewire 2 to Livewire 3.5 and I'm having multiple issues with it. One of which is that I have some inline JS scripts that don't work anymore.

For context, I have a repeater element where I can add new instances of a component dynamically. This component has a script in it, like this: 

```php
        @foreach ($exceptions as $index => $exception)
            <x-options-form.exceptionelement :index="$index" :exception="$exception" />

                @script
                    <script>
                        console.log('hi!')
                    </script>
                @endscript
        @endforeach
```

I tried:
- adding `@stack('scripts')` or `@stack('js')` to my layout file's head
- adding the index as a `wire:key` to the top element of the component
- moving the script directly in the loop instead of in the component (like above)

The script only executes when loading the page and there are already elements in the repeater, but not when adding new ones. Any idea what could be going on? Thank you 😊

**Comment by morceaudebois** [comment]
Alright, I kind of manage to deal with this using Alpine's `x-init` inside the component itself:

```php
<x-all.text-input   
    {{ $attributes }}
    type="text"
    wire:model="{{ $attributes['wire:model'] }}"
    x-init="flatpickr($el, {
        enableTime: true,
        noCalendar: true,
        time_24hr: true
    })"
/>
```

It works but I'll admit I'd prefer to be able to use regular inline scripts. This won't even work as the function apparently isn't defined:

```php
<x-all.text-input   
    {{ $attributes }}
    type="text"
    wire:model="{{ $attributes['wire:model'] }}"
    x-init="sayHi()"
/>

@script
    <script>
        function sayHi() {
            console.log('hi!')
        }
    </script>
@endscript
```

Would still appreciate some insight from the pros!

**Reply by danie-ramdhani** [reply, marked as answer]
you need to make the function inside `@script` global

```blade
<div x-data="sayHi()">
    {{-- ... --}}
</div>

@script
    <script>
        window.sayHi = () => {
            alert('hi')
        }
    </script>
@endscript
```

or you can use `Alpine.data`

```blade
<div x-data="sayHi()">
    {{-- ... --}}
</div>

@script
    <script>
        Alpine.data('sayHi', () => ({
            init() {
                alert('hi')
            }
        }));
    </script>
@endscript
```

learn more: https://alpinejs.dev/globals/alpine-data

**Reply by morceaudebois** [reply]
Right! Thanks for that.

I think I'll mark it as answered, this is a bit more of a workaround rather than a real solution for my initial problem, but I think it should work in most similar situations.
